### ISSUE TITLE
Support value function prematurely returns zero

### ISSUE THREAD

**Issue by ueliwechsler** [JuliaReach/LazySets.jl#1644]
When running the support value function `ρ` on small vectors, the return value is jumps to zero for a small change in the input vector.
```julia
a = rand(HPolytope)
ρ([11.0e-8,1.9e-8], a) # 7.02e-8
ρ([10.0e-8,1.9e-8], a) # 0.00
```
This results in bad or wrong convergence properties for some algorithms.
It would be nice to, either make it possible to define the tolerance for this function manually or make the tolerance in general lower, such that it returns non-zero results up to `eps(Flaot64)` or some similar small value.

**Comment by schillic** [comment]
Thanks for the small example - it was helpful for investigating :+1:

We evaluate the support function by solving an LP. I tracked the issue down to the LP solver. These are the LPs we create:
```julia
using GLPKMathProgInterface, MathProgBase

d = [0.0, -1.9e-8];
A = [-0.219301 -0.0547087; 0.00570639 0.282715; 0.0638717 -0.0670924; 0.144842 0.023805];
b = [0.120026, 0.0329985, 0.0945786, 0.069134];

for d1 in [-1.1e-7, -1.0e-7]
    d[1] = d1
    println(linprog(d, A, '<', b, -Inf, Inf, GLPKSolverLP()))
end
```
Output:
```julia
MathProgBase.HighLevelInterface.LinprogSolution(:Optimal,
 -5.2503693680009945e-8, [0.477306, 0.0],
 Dict{Any,Any}(:lambda=>[0.0, 0.0, 0.0, -7.59448e-7],:redcost=>[0.0, -9.21335e-10]))
MathProgBase.HighLevelInterface.LinprogSolution(:Optimal,
 0.0, [0.0, 0.0],
 Dict{Any,Any}(:lambda=>[0.0, 0.0, 0.0, 0.0],:redcost=>[-1.0e-7, -1.9e-8]))
```
So I guess that we cannot do much here because this seems to be the precision boundary of the solver.

### Workarounds
1. As long as you operate in 2D, you can `convert` to `HPolygon`, for which we do not solve an LP.
2. You can normalize the direction vector to get to a better number range:
```julia
using LazySets, Polyhedra

P = rand(HPolytope);
Q = convert(HPolygon, P);
for d in [[11.0e-8,1.9e-8], [10.0e-8,1.9e-8]]
    println("\nd = $d")
    println(ρ(d, P))
    println(ρ(d, Q))
    println(ρ(normalize(d), P) * norm(d))
end
```
Output:
```julia
d = [1.1e-7, 1.9e-8]
1.643926928700733e-7
1.823817774848231e-7
1.823817774848231e-7

d = [1.0e-7, 1.9e-8]
0.0
1.6545193553280092e-7
1.6545193553280092e-7
```
We might implement workaround 2. as an option for the support function, but for now I suggest you just define your own helper function.

**Comment by ueliwechsler** [comment]
I did play around a bit more and as expected, the same problem occurs with the support vector function `σ`. But, it did not happen in all cases. For example, if I have a `Hyperectangle` in general or an `HPolytope` that contains the origin in its interior, the right values for the support functions were returned.

Is there any drawback to use ```ρ(normalize(d), P) * norm(d))```and  ```σ(normalize(d), P)``` apart from performance?

 Concerning the performance, the adjusted functions have 3, respectively 2, allocations more. But it is not necessarily slower, e.g.
```
  25.527 μs (63 allocations: 4.23 KiB)
  24.433 μs (62 allocations: 4.14 KiB)
``` 
for the example below. See this example as a case, where the support vector function also returns the wrong result for small direction vectors.

```julia
using Plots
using Lazysets
using LinearAlgebra
A = [0.713204 -0.276325;
    -0.00879969 -0.110889;
    0.0256732 0.00329082;
    -0.0126709 0.014486]
b = [-0.186085;
    0.124179;
    0.0112762;
    0.0185422]
P = HPolytope(A,b)
d = 1e-8*[3; 0.1]
plot(P)
σ_norm = σ(normalize(d), P)
σ_reg = σ(d, P)
plot!(LineSegment(zeros(2),normalize(d)))
plot!(LineSegment(zeros(2),σ_norm))
plot!(LineSegment(zeros(2),σ_reg))
```
![image](https://user-images.githubusercontent.com/12156892/64111655-7e9be080-cd85-11e9-9f35-c854e07a6bc9.png)

**Comment by mforets** [comment]
> Is there any drawback to use ρ(normalize(d), P) * norm(d))and σ(normalize(d), P) apart from performance?

I dont think so; they are equivalent [by property (1.1) here](https://juliareach.github.io/LazySets.jl/latest/man/polyhedral_approximations/#Support-Function-1).

**Comment by mforets** [comment]
Thanks for sharing the example and the plot. Maybe we could add a `normalize` kwarg for the `HPolytope` supp func that performs the operation you've described. On the longer run we could also try to use consistently the constants defined in `comparisons.jl` (eg. to pass such tolerances to the solver).

**Comment by ueliwechsler** [comment]
I think we could do both things. The support function with the `normalized` input could still be solved with the tolerances defined in `comparison.jl`.

**Comment by mforets** [comment]
There could also be a function to `normalize` the polytope, cf.  http://people.ee.ethz.ch/~mpt/2/docs/refguide/mpt/@polytope/normalize.html.

**Comment by schillic** [comment]
> There could also be a function to `normalize` the polytope, cf. http://people.ee.ethz.ch/~mpt/2/docs/refguide/mpt/@polytope/normalize.html.

The polytope was "normal" but the direction was "abnormal".

**Comment by mforets** [comment]
:+1: True, that is something else. I opened #1698.
